infra/nightly-resources/index.js

```diff
@@ -30,254 +30,22 @@ function clearWarnings() {
-
-  var name = current.runMethod;
-  if (current.llvm_ir) {
-    name = `<a target="_blank" rel="noopener noreferrer" href="llvm.html?benchmark=${benchName}&runmode=${current.runMethod}">${current.runMethod}</a>`;
-  }
-
-  const result = {
-    name: name,
-    mean: { class: "", value: tryRound(results.mean) },
-    meanVsBaseline: diffAttribute(results, baselineResults, "mean"),
-    min: { class: "", value: tryRound(results.min) },
-    minVsBaseline: diffAttribute(results, baselineResults, "min"),
-    max: { class: "", value: tryRound(results.max) },
-    maxVsBaseline: diffAttribute(results, baselineResults, "max"),
-    median: { class: "", value: tryRound(results.median) },
-    medianVsBaseline: diffAttribute(results, baselineResults, "median"),
-    stddev: { class: "", value: tryRound(results.stddev) },
-  };
-
-  return result;
-}
-
 function refreshView() {
-  const parsed = Array.from(GLOBAL_DATA.enabledBenchmarks).map((benchName) => {
-    if (!GLOBAL_DATA.baselineRun) {
-      addWarning("no baseline to compare to");
-    }
-    const executions = Object.keys(GLOBAL_DATA.currentRun[benchName])
-      .map((runMode) => {
-        // if the mode is not enabled, skip it
-        if (!GLOBAL_DATA.enabledModes.has(runMode)) {
-          return undefined;
-        }
-        // prevRun may be undefined
-        const baselineBench = GLOBAL_DATA.baselineRun?.[benchName];
-        if (GLOBAL_DATA.baselineRun && !baselineBench) {
-          addWarning(`Baseline doesn't have ${benchName} benchmark`);
-        }
-        const baselineRunForMethod = baselineBench?.[runMode];
-        if (baselineBench && !baselineRunForMethod) {
-          addWarning(
-            `Baseline doesn't have run mode ${runMode} for ${benchName}`,
-          );
-        }
-
-        return buildEntry(
-          benchName,
-          baselineRunForMethod,
-          GLOBAL_DATA.currentRun[benchName][runMode],
-        );
-      })
-      .filter((e) => e !== undefined);
-    if (executions.length > 1) {
-      const cols = ["mean", "min", "max", "median"];
-      cols.forEach((col) => {
-        const sorted = executions
-          .map((e) => e[col])
-          .sort((a, b) => a.value - b.value);
-        const min = sorted[0].value;
-        const max = sorted[sorted.length - 1].value;
-        sorted.forEach((item) => {
-          if (item.value === min) {
-            item.class = "good";
-          }
-          if (item.value === max) {
-            item.class = "bad";
-          }
-        });
-      });
-    }
-
-    return {
-      name: benchName,
-      executions: { data: executions },
-    };
-  });
+  if (!GLOBAL_DATA.baselineRun) {
+    addWarning("no baseline to compare to");
+  }
 
-  parsed.sort((l, r) => {
-    if (l.name < r.name) {
-      return -1;
-    }
-    if (l.name > r.name) {
-      return 1;
-    }
-    return 0;
+  const byBench = {};
+  Array.from(GLOBAL_DATA.enabledBenchmarks).forEach((benchmark) => {
```

Comment:
Why do we need `Array.from`?

Reply:
It's a `Set`

Reply:
Actually, not needed, because order of iteration won't matter here.